ownership-chain/pallets/living-assets-ownership/src/lib.rs

```diff
@@ -218,62 +210,65 @@ pub enum CollectionError {
 	InvalidPrefix,
 }
 
-/// Converts a `CollectionId` into an `H160` address format.
+/// Converts a `CollectionId` into an `Address`.
 ///
 /// This function takes the given `CollectionId`, which is assumed to be a `u64`,
-/// and maps it into an `H160` address, prepending it with the `ASSET_PRECOMPILE_ADDRESS_PREFIX`.
+/// and maps it into an `Address` address, prepending it with the `ASSET_PRECOMPILE_ADDRESS_PREFIX`.
 ///
 /// # Arguments
 ///
 /// * `collection_id`: The ID of the collection to be converted.
 ///
 /// # Returns
 ///
-/// * An `H160` representation of the collection ID.
-pub fn collection_id_to_address(collection_id: CollectionId) -> H160 {
+/// * An `Address` representation of the collection ID.
+pub fn collection_id_to_address<Address: From<[u8; 20]>>(collection_id: CollectionId) -> Address {
 	let mut bytes = [0u8; 20];
 	bytes[12..20].copy_from_slice(&collection_id.to_be_bytes());
 	for (i, byte) in ASSET_PRECOMPILE_ADDRESS_PREFIX.iter().enumerate() {
 		bytes[i] = *byte;
 	}
-	H160(bytes)
+	Address::from(bytes)
 }
 
-/// Converts an `H160` address into a `CollectionId` format.
+/// Converts an `Address` address into a `CollectionId` format.
 ///
-/// This function takes the given `H160` address, checks for the correct prefix, and extracts
+/// This function takes the given `Address` address, checks for the correct prefix, and extracts
 /// the `CollectionId` from it. If the prefix is incorrect, it returns a
 /// `CollectionError::InvalidPrefix` error.
 ///
 /// # Arguments
 ///
-/// * `address`: The `H160` address to be converted.
+/// * `address`: The `Address` address to be converted.
 ///
 /// # Returns
 ///
 /// * A `Result` which is either the `CollectionId` or an error indicating the address is invalid.
-pub fn address_to_collection_id(address: H160) -> Result<CollectionId, CollectionError> {
-	if &address.0[0..12] != ASSET_PRECOMPILE_ADDRESS_PREFIX {
+pub fn address_to_collection_id<Address: Into<[u8; 20]>>(
```

Comment:
added this type param because theses functions were using `H160` which is not the same type as `fp_account::EthereumSigner`, although inner value is the same (`[u8;20]`) and it causes type conflicts if not changed